src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java

```diff
@@ -0,0 +1,86 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.List;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("Exploded Blocks")
+@Description("Get all the blocks that were destroyed in an explode event")
+@Examples({"on explode:",
+	"\tloop exploded blocks:",
+	"\t\tadd loop-block to {exploded::blocks::*}"})
+@Events("explode")
+@Since("INSERT VERSION")
+public class ExprExplodedBlocks extends SimpleExpression<Block> {
+
+	static {
+		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.COMBINED, "[the] (destroyed|exploded) blocks");
+	}
+	
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		if (!ScriptLoader.isCurrentEvent(EntityExplodeEvent.class)) {
+			Skript.error("Exploded blocks can only be retrieved from an explode event.");
+			return false;
+		}
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	protected Block[] get(Event e) {
+		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
+		return blockList.toArray(new Block[blockList.size()]);
```

Comment:
`return block.toArray(new Block[0]);` is actually faster

https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html

Reply:
The difference is quite small, and relies on JIT compiler optimization that might or might not have it. Hotspot's C2 does have it, but what about C1, Graal or OpenJ9?

I guess we could also argue that even non-intrinsified call is ok, because this is not performance-critical code.